gts: Fix cross compilation

kampka opened the pull request:
###### Motivation for this change

gts generate binaries during the build process that are used during the
build process. When building cross-platform, these binaries are not
executable as they are built with for the target platform instead of the
host platform. The boostrap build introduced here will first build gts
natively and then cross-platform, using the native binaries in the
cross-build.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

kampka commented:
@GrahamcOfBorg build gts
@GrahamcOfBorg build pkgsCross.aarch64-multiplatform.gts

jonringer commented:
@GrahamcOfBorg eval

jtojnar commented on pkgs/development/libraries/gts/default.nix:
```diff
@@ -1,34 +1,65 @@
+      sha256 = "07mqx09jxh8cv9753y2d2jsv7wp8vjmrd7zcfpbrddz3wc9kx705";
+    };
+
+    # glib needs to be in nativeBuildInputs in order for AM_PATH_GLIB_2_0
```
This is suspicious, I would expect  the macros to be picked up just like pkg-config files are.

kampka replied:
The macro itself is not the issue, it's the AC_TRY_RUN that is used by AM_PATH_GLIB_2_0 internally which will fail if not run in native context.

jtojnar commented on pkgs/development/libraries/gts/default.nix:
```diff
@@ -1,34 +1,65 @@
-}
+
+  bootstrap = stdenv.mkDerivation (drv { pname = "${name}-bootstrap"; } // stdenv.lib.optionalAttrs (cross) {
+    configureFlags = [ "CC=${buildPackages.stdenv.cc}/bin/cc" ];
```
Would not it perhaps be possible to just change the compiler for building the `predicates_init` in the `Makefile.am`? I am not very familiar with automake myself but this looks overly complicated.

kampka committed:
gts: Fix cross compilation

gts generate binaries during the build process that are used during the
build process. When building cross-platform, these binaries are not
executable as they are built with for the target platform instead of the
host platform. The boostrap build introduced here will first build gts
natively and then cross-platform, using the native binaries in the
cross-build.

kampka commented:
@GrahamcOfBorg build gts
@GrahamcOfBorg build pkgsCross.aarch64-multiplatform.gts
